crates/core/src/models/metadata.rs

```diff
@@ -0,0 +1,10 @@
+use reth_primitives::U256;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct TokenMetadata {
+    pub decimals: Option<U256>,
+    pub name: Option<String>,
```

Comment:
Should these fields be String or &str? 
It's an open question, I'm not yet aware of the trade-offs associated

Reply:
My understanding is that if we don't know the contents of the string at runtime, it makes more sense to use `String`. Also using `&str` would involve setting up unnecessary lifetimes in this case.

Reply:
Works for me!